examples/packages/single-language/python-project/default.nix

```diff
@@ -12,7 +12,7 @@ in {
   ];
 
   deps = {nixpkgs, ...}: {
-    python = nixpkgs.python310;
+    python = nixpkgs.${pyproject.tool.dream2nix.python-version};
```

Comment:
I think we could parse the python version directly from the `requires_python` field using [pyproject.nix](https://github.com/nix-community/pyproject.nix).
But maybe this can be done in a separate PR.
If you are interested in working on this, you can see how to access the pyproject.nix library if you look at the `WIP-python-pdm` module as an example that already uses this lib (it doesn't parse yet the python version, though).